### PR TITLE
NBG7815

### PR DESCRIPTION
**Pull request by itorK**
Zyxel Armor G5 NBG7815 is a AX WIFI router with 4 1G and 2 10G ports.

Specifications:

    •     CPU: Qualcomm IPQ8074A Quad core Cortex-A53 2.2GHz
    •     RAM: 1024MB of DDR3
    •     Storage: 4GB eMMC (contains kernel and rootfs) / 8MB NOR
          (contains art and u-boot-env)
    •     Ethernet: 4x 1G RJ45 ports + 2 10G ports (Aquantia AQR107)
    •     WLAN:
          2.4GHz: Qualcomm QCN5024 2x2 802.11b/g/n/ax 1174 Mbps PHY rate
          5GHz: Qualcomm QCN5054 4x4 802.11a/b/g/n/ac/ax 2402 PHY rate

    NOT WORKING
        •     LEDs controlled by I2C LP5562 - (drivers not in linux kernel)

        A poulated serial header is onboard.
        RX/TX is working, bootwait is active, secure boot is not enabled.
    
       SSH and root access:
      
        https://openwrt.org/inbox/toh/zyxel/nbg7815_armor_g5

        Forum:

        https://forum.openwrt.org/t/openwrt-support-for-armor-g5-nbg7815
       
        Installation Instructions:
            • connect Serial Interface
            • stop auto boot
            • setenv serverip 192.168.10.1
            • setenv ipaddr 192.168.10.10
            • tftpboot the initramfs image
              (openwrt-ipq807x-generic-nbg7815-initramfs-fit-uImage.itb)
            • bootm

Signed-off-by: Karol Przybylski <itor@o2.pl>

### CODE REVIEW

**Comment by robimarko** [comment]
Why closed?

**Comment by itorK** [comment]
my fault, so reopen :)

**Comment by robimarko** [comment]
You need to rebase this and also squash the development changes you have

**Comment by joba-1** [comment]
minor question:

first line says `... 2 10G ports.`
forth specification item says `... 1 10G ports`

what is it?

**Comment by robimarko** [comment]
No, what are you doing with this squash commit?

**Comment by itorK** [comment]
> No, what are you doing with this squash commit?

Now it's ok for you?

**Comment by robimarko** [comment]
No, now it's even worse.

You need to remove these 2 newest commits and then rebase the whole thing on my current branch.
You basically need to be left with one commit adding support for the device before I even look at it.

**Comment by rmandrad** [comment]
@itorK don't add anything else (other devices, patches unrelated to this device) than what is related to your changes for this device, nothing else!  basically a feature that you want to implement ... a combination of commits/code  into one.

here's a guide - https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits & another one more hands-on https://www.golinuxcloud.com/git-squash-commits/

**Comment by itorK** [comment]
Now, should be okay, it's only 1 commit with changes NBG7815 only

File: target/linux/ipq807x/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -30,6 +30,9 @@ ipq807x_setup_interfaces()
 	xiaomi,ax9000)
 		ucidef_set_interfaces_lan_wan "eth0 eth1 eth2 eth3" "eth4"
 		;;
+	zyxel,nbg7815)
```
Comment (robimarko) [inline target/linux/ipq807x/base-files/etc/board.d/02_network:33]: Doesn't Qnap use the same network config?

File: package/boot/uboot-envtools/files/ipq807x [hunk 2]
```diff
@@ -38,6 +38,11 @@ qnap,301w)
 	[ -n "$idx" ] && \
 		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x20000" "0x20000" "1"
 	;;
+zyxel,nbg7815)
```
Comment (robimarko) [inline package/boot/uboot-envtools/files/ipq807x:41]: EAP102 should be using the same config, no need for duplication

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Zyxel NBG7815";
+	compatible = "zyxel,nbg7815", "qcom,ipq8074";
+  
+	aliases {
+		serial0 = &blsp1_uart5;
+		/* LED is managed by lp5569
+    led-boot = &led_system_yellow;
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:20]: Broken indent

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+		led-running = &led_system_blue;
+		led-upgrade = &led_system_yellow;
+    */
+		/* Aliases as required by u-boot to patch MAC addresses */
+		ethernet0 = &dp1;
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5;
+    ethernet5 = &dp6;
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:31]: Broken indent

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+		label-mac-device = &dp1;
+	};
+
+	chosen {
+    stdout-path = "serial0:115200n8";
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:36]: Broken indent again

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+		bootargs = "console=ttyMSM0,115200n8 root=PARTUUID=a0bb3852-36cc-ac69-a1a2-6bd39ff56f0a rootwait";
+		bootargs-append = " swiotlb=1";
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:38]: What is `bootargs-append` used for?

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+	};
+  
+  
+  keys {
+		compatible = "gpio-keys";
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:43]: Broken indents

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+		reset {
+			label = "reset";
+			gpios = <&tlmm 36 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&tlmm {
+  mdio_pins: mdio-pins {
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:54]: Broken indents

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+
+	qca8075_3: ethernet-phy@3 {
+		reg = <3>;
+	};
+
+	qca8075_4: ethernet-phy@4 {
+		reg = <0x1c>;
+	};
+  
+  qca8075_5: ethernet-phy@5 {
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:118]: Broken indent

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 10]
```diff
@@ -0,0 +1,517 @@
...
+&qpic_bam {
+  status = "okay";
+  };
+
+&mdio {
+	status = "okay";
+
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 37 GPIO_ACTIVE_LOW>; /*<0x14 0x25 0x00 0x14 0x1f 0x01>;*/
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:96]: Remove the comment

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts (hunk 9 above)
Comment (robimarko) [reply target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:118]: Also, this is the AQR?

If so, label you are using is completely misleading, you are also missing its reset GPIO

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 11]
```diff
@@ -0,0 +1,517 @@
...
+    compatible = "ethernet-phy-ieee802.3-c45";
+    reg = <0x08>;
+  };  
+
+};
+
+&switch {
+	status = "okay";
+  
+  switch_cpu_bmp = <0x01>;
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:128]: Again, completely broken indents

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 12]
```diff
@@ -0,0 +1,517 @@
...
+	qca8075_2: ethernet-phy@2 {
+		reg = <2>;
+	};
+
+	qca8075_3: ethernet-phy@3 {
+		reg = <3>;
+	};
+
+	qca8075_4: ethernet-phy@4 {
+		reg = <0x1c>;
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:115]: Is this the 2.5G PHY?

If so, label is completely misleading and you are missing its reset GPIO

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 13]
```diff
@@ -0,0 +1,517 @@
...
+				};
+			};
+		};
+	};
+};
+  
+&edma {
+	status = "okay";
+};
+
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:449]: Extra newline

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 14]
```diff
@@ -0,0 +1,517 @@
...
+  
+&soc {  
+  dp1: dp1 {
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:452]: Broken indent

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 15]
```diff
@@ -0,0 +1,517 @@
...
+			device_type = "network";
+			compatible = "qcom,nss-dp";
+			qcom,id = <1>;
+			reg = <0x3a001000 0x200>;
+			qcom,mactype = <0>;
+			local-mac-address = [bc cf 4f ff 07 10];
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:458]: Dont harcode the MAC in DTS

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 16]
```diff
@@ -0,0 +1,517 @@
...
+			qcom,phy-mdio-addr = <0x00>;
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:459]: This has been dropped completely way ago, was it even working on recent versions?
Use `phy-handle` to pass the PHY instead

File: target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts [hunk 17]
```diff
@@ -0,0 +1,517 @@
...
+			device_type = "network";
+			compatible = "qcom,nss-dp";
+			qcom,id = <6>;
+			reg = <0x3a007000 0x3fff>;
+			qcom,mactype = <1>;
+			local-mac-address = [00 00 00 00 00 00];
+			qcom,phy-mdio-addr = <0x08>;
+			phy-mode = "sgmii";
+		};
+	};
```
Comment (robimarko) [inline target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts:517]: Missing newline at the end

File: target/linux/ipq807x/base-files/etc/board.d/02_network (hunk 1 above)
Comment (rmandrad) [reply target/linux/ipq807x/base-files/etc/board.d/02_network:33]: no '"ucidef_set_interfaces_lan_wan "eth0 eth1 eth2 eth3 eth4" "eth5"'

Comment (itorK) [reply target/linux/ipq807x/base-files/etc/board.d/02_network:33]: Yes, looks similar.
Original Zyxel NBG7815 configuration is here:
https://github.com/kirdesde/nbg7815_gpl/blob/e816d72c67037186f2d5c10639f609649cbcce10/qca/src/linux-4.4/arch/arm64/boot/dts/qcom/qcom-ipq807x-nbg7815.dts

Comment (robimarko) [reply target/linux/ipq807x/base-files/etc/board.d/02_network:33]: Not just similar, but unless I am blind its exactly the same.
So, please confirm that and if so, there is no need for duplication.

**Comment by robimarko** [comment]
Please keep the changes squashed, its hard to review like this.

**Comment by robimarko** [comment]
Again, please rebase and squash the changes, please keep everything squashed in one commit, its hard to see review otherwise

**Pull request closed without merging**